java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -205,9 +218,25 @@ public AccessToken refreshAccessToken() throws IOException {
             }
 
             if (shouldRefresh(this.currentToken, clock.instant())) {
+                String lockId = "oauth_refresh_token";
+                boolean acquired = lockService.acquire(lockId);
+
+                if (!acquired) {
+                    try {
+                        Thread.sleep(attempt * 2_000L);
```

Comment:
guava Uninterruptibles, we shouldn't break execution if that happens
```suggestion
Uninterruptibles.sleepUninterruptibly(attempt * 2_000L, TimeUnit.MILLISECONDS);
```